DataManagementSystem/Client/ConsistencyChecks.py

```diff
@@ -0,0 +1,836 @@
+        if not replicasRes['OK']:
+          gLogger.error( "error:  %s" % replicasRes['Message'] )
+          raise RuntimeError( "error:  %s" % replicasRes['Message'] )
+        replicasRes = replicasRes['Value']
+        if replicasRes['Failed']:
+          retDict['NoReplicas'].update( replicasRes['Failed'] )
+        replicas.update( replicasRes['Successful'] )
+      self.__write( ' (%.1f seconds)\n' % ( time.time() - startTime ) )
+
+    self.__write( "Get FC metadata for %d files to be checked: " % len( lfns ) )
+    metadata = {}
+    for lfnChunk in breakListIntoChunks( replicas.keys(), chunkSize ):
+      self.__write( '.' )
+      res = self.fc.getFileMetadata( lfnChunk )
+      if not res['OK']:
+        raise RuntimeError( "error %s" % res['Message'] )
+      metadata.update( res['Value']['Successful'] )
+    self.__write( ' (%.1f seconds)\n' % ( time.time() - startTime ) )
+
+    gLogger.always( "Check existence and compare checksum file by file..." )
+    csDict = {}
+    seFiles = {}
+    surlLfn = {}
+    startTime = time.time()
+    # Reverse the LFN->SE dictionary
+    for lfn in replicas:
+      csDict.setdefault( lfn, {} )[ 'LFCChecksum' ] = metadata.get( lfn, {} ).get( 'Checksum' )
+      replicaDict = replicas[ lfn ]
+      for se in replicaDict:
+        surl = replicaDict[ se ]
+        surlLfn[surl] = lfn
+        seFiles.setdefault( se, [] ).append( surl )
+
+    checkSum = {}
+    self.__write( 'Getting checksum of %d replicas in %d SEs (chunks of %d): ' % ( len( surlLfn ), len( seFiles ), chunkSize ) )
+    pfnNotAvailable = {}
+    logLevel = gLogger.getLevel()
+    gLogger.setLevel( 'FATAL' )
+    for num, se in enumerate( sorted( seFiles ) ):
+      self.__write( '\n%d. At %s (%d files): ' % ( num, se, len( seFiles[se] ) ) )
+      oSe = StorageElement( se )
+      for surlChunk in breakListIntoChunks( seFiles[se], chunkSize ):
+        self.__write( '.' )
+        surlRes = oSe.getFileMetadata( surlChunk )
+        if not surlRes['OK']:
+          gLogger.error( "error StorageElement.getFileMetadata returns %s" % ( surlRes['Message'] ) )
+          raise RuntimeError( "error StorageElement.getFileMetadata returns %s" % ( surlRes['Message'] ) )
+        surlRes = surlRes['Value']
+        for surl in surlRes['Failed']:
+          lfn = surlLfn[surl]
+          gLogger.info( "SURL was not found at %s! %s " % ( se, surl ) )
+          pfnNotAvailable.setdefault( lfn, [] ).append( se )
+        for surl in surlRes['Successful']:
+          lfn = surlLfn[surl]
+          checkSum.setdefault( lfn, {} )[se] = surlRes['Successful'][ surl ]['Checksum']
+    self.__write( ' (%.1f seconds)\n' % ( time.time() - startTime ) )
+    gLogger.setLevel( logLevel )
+    retDict[ 'MissingPFN'] = {}
+
+    startTime = time.time()
+    self.__write( 'Verifying checksum of %d files (chunks of %d) ' % ( len( replicas ), chunkSize ) )
+    for num, lfn in enumerate( replicas ):
+      # get the lfn checksum from the LFC
```

Comment:
There is no lfc anymore. Maybe renaming it into a more generic "file catalog checksum" is more appropriate ?